alerts/cluster-monitoring-operator/PrometheusOperatorRejectedResources.md

```diff
@@ -0,0 +1,88 @@
+
+## Meaning
+
+The `PrometheusOperatorRejectedResources` alert triggers when Prometheus Operator
+rejects  `Alertmanager` `PodMonitor`, `ServiceMonitor`, or `PrometheusRules` objects
+because custom resources in an object are not configured correctly.
+
+## Impact
+
+The custom resources that trigger the alert are ignored by Prometheus Operator.
+As a consequence, they will not be part of the final configuration of the
+Prometheus, Alertmanager, or Thanos Ruler components managed by Prometheus Operator.
+
+## Diagnosis
+
+Find the type of custom resource and its namespace in the `description` field
+of the alert `PrometheusOperatorRejectedResources`.
+
+By CLI
+```bash
+### Gets alertmanager URL
+$ ALERT_MANAGER=$(oc get route alertmanager-main -n openshift-monitoring -o jsonpath='{@.spec.host}')
+
+### Gets fired alerts
+$ curl -k -H "Authorization: Bearer $(oc create token prometheus-k8s -n openshift-monitoring)"  https://$ALERT_MANAGER/api/v1/alerts
+```
+
+In Openshift web console:
+Browse to the tab Observe -> Alerting, then search the alert `PrometheusOperatorRejectedResources`
+in firing alerts list.
+![console alert list](PrometheusOperatorRejectedResources-alert-list.png)
+
+Click on the alert to see its details and scroll down to see its `Description` field.
+The namespace should be displayed as a link, followed by the resource type.
+
+![console alert detail](PrometheusOperatorRejectedResources-alert-detail.png)
+
+Then check recently modified custom resource of that type in that namespace.
+If the namespace is named `openshift-.*` except `openshift-user-workload-monitoring`,
+for example, `openshift-monitoring`, this may be a bug in Openshift,
+please file a support case.
+
+For more details of the error, check logs of Prometheus Operator pod in
+the namespace `openshift-user-workload-monitoring`.
+```bash
+# Get the name of the Prometheus Operator pod
+POD=$(oc get pod -n openshift-user-workload-monitoring | grep -Eo 'prometheus-operator-[0-9a-z]*-[0-9a-z]*')
+
+# Check its logs
+oc logs $POD -n openshift-user-workload-monitoring
+```
+
+Here is a typical error message about rejected custom resource.
+```log
+level=warn ts=2023-07-03T20:37:20.740723141Z caller=operator.go:1917 component=prometheusoperator msg="skipping servicemonitor" error="it accesses file system via bearer token file which Prometheus specification prohibits" servicemonitor=quarkus-demo/otel-collector namespace=openshift-user-workload-monitoring prometheus=user-workload
+```
+
+## Mitigation
+
+Fix the misconfigured custom resource and reapply it to the cluster.
+
+The custom resource itself should already conform CRD schema, otherwise the API server
+will reject it before the Prometheus Operator does.
+
+The error often comes from the value of certain fields that the operator cannot
+apply it to make a valid configuration. The causes for this alert can vary
+depending on the specific configuration and deployment of Prometheus Operator.
```

Comment:
```suggestion
This error often occurs because the Operator cannot apply a value from certain fields 
to create a valid configuration. The causes for this alert can vary
depending on the specific configuration and deployment of Prometheus Operator.
```